File: rust/src/options.rs
/*
 * -*- coding: utf-8 -*-
 *
 * jomiel-examples
 *
 * Copyright
 *  2019 Toni Gündoğdu
 *
 *
 * SPDX-License-Identifier: Apache-2.0
 */

use structopt::StructOpt;

#[derive(StructOpt, Debug)]
#[structopt(name = "demo")]
pub struct Options {
    /// Specify maximum time in seconds for the connection allowed to take
    #[structopt(
        name = "time",
        short = "t",
        long = "connect-timeout",
        default_value = "30"
    )]
    pub connect_timeout: i64,

    /// Specify the router endpoint address
    #[structopt(
        name = "addr",
        short = "r",
        long = "router-endpoint",
        default_value = "tcp://localhost:5570"
    )]
    pub router_endpoint: String,

    /// Print configuration values and exit
    #[structopt(short = "D", long = "print-config")]
    pub print_config: bool,

    /// Display ZeroMQ version and exit
    #[structopt(short = "V", long = "version-zmq")]
    pub version_zmq: bool,

    /*
    /// Print dumped messages in JSON
    #[structopt(short = "j", long = "output-json")]
    pub output_json: bool,
    */
    /// Be brief and to the point; dump interesting details only
    #[structopt(short = "q", long = "be-terse")]
    pub be_terse: bool,

    /// the URIs to parse
    #[structopt()]
    pub uri: Vec<String>,
}

File: nodejs/src/options.js
/*
 * -*- coding: utf-8 -*-
 *
 * jomiel-examples
 *
 * Copyright
 *  2021 Toni Gündoğdu
 *
 *
 * SPDX-License-Identifier: Apache-2.0
 */

const USAGE = `
Usage:
    demo [-hDVjcq] [-r <addr>] [-t <time>] [URI ...]

Options:
     -h --help                      Print this help and exit
     -D --print-config              Print configuration values and exit
     -V --version-zmq               Print ZeroMQ version and exit
     -j --output-json               Print dumped messages in JSON
     -c --compact-json              Use more compact representation of JSON
     -r --router-endpoint <addr>    Specify the router endpoint address
                                     [default: tcp://localhost:5514]
     -t --connect-timeout <time>    Specify maximum time in seconds for
                                     the connection allowed to take
                                     [default: 30]
     -q --be-terse                  Be brief and to the point; dump
                                     interesting details only
`;

const parse = () => {
  return require("docopt").docopt(USAGE);
};

module.exports = { parse };
